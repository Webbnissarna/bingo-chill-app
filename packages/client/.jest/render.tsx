import React from "react";
import { render as rtlRender } from "@testing-library/react";

export function render(ui: React.ReactElement) {
  return rtlRender(<>{ui}</>);
}
