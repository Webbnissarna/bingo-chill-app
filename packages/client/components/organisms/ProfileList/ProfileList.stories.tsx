import { SPOOKY_GHOST_IMAGE_BASE64 } from "@/.ladle/constants";
import ProfileList from ".";
import type { StoryDefault } from "@ladle/react";

export function Default() {
  return (
    <div className="bg-polarNight-1 p-5">
      <ProfileList
        profiles={[
          {
            id: "a",
            icon: SPOOKY_GHOST_IMAGE_BASE64,
            badgeValue: 0,
            trimColor: "#bf616a",
          },
          {
            id: "b",
            icon: SPOOKY_GHOST_IMAGE_BASE64,
            badgeValue: 1,
            trimColor: "#a3be8c",
          },
          {
            id: "c",
            icon: SPOOKY_GHOST_IMAGE_BASE64,
            badgeValue: 2,
            trimColor: "#b48ead",
          },
          {
            id: "d",
            icon: SPOOKY_GHOST_IMAGE_BASE64,
            badgeValue: 3,
            trimColor: "#ebcb8b",
          },
          {
            id: "e",
            icon: SPOOKY_GHOST_IMAGE_BASE64,
            badgeValue: 4,
            trimColor: "#d08770",
          },
        ]}
      />
    </div>
  );
}

export default {
  title: "Organisms / Profile List",
} satisfies StoryDefault;
