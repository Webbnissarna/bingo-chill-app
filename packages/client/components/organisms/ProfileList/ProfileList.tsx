import ProfileIcon from "@/components/molecules/ProfileIcon/ProfileIcon";

export interface Profile {
  id: string;
  name: string;
  icon: string;
  badgeValue: number;
  trimColor: string;
}

interface ProfileListProps {
  profiles: Profile[];
}

export default function ProfileList({
  profiles,
}: ProfileListProps): JSX.Element {
  return (
    <div className="flex gap-6 p-4 flex-wrap-reverse scrollbar-thin">
      {profiles.map((profile) => (
        <div key={profile.id}>
          <ProfileIcon
            name={profile.name}
            icon={profile.icon}
            badgeValue={profile.badgeValue}
            trimColor={profile.trimColor}
          />
        </div>
      ))}
    </div>
  );
}
