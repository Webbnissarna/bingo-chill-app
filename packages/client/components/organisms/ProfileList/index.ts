export { default } from "./ProfileList";
export * from "./ProfileList";
