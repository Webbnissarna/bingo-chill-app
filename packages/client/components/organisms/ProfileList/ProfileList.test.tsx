import { render } from "../../../.jest/render";
import ProfileList from ".";

describe("ProfileList", () => {
  it("renders", () => {
    const { asFragment } = render(<ProfileList profiles={[]} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
