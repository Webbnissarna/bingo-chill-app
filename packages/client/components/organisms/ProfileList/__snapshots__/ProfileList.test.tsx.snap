// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileList renders 1`] = `
<DocumentFragment>
  <div
    class="flex gap-6 p-4 flex-wrap-reverse scrollbar-thin"
  />
</DocumentFragment>
`;
